pandas/core/computation/expr.py

```diff
@@ -39,10 +36,125 @@
 
 import pandas.io.formats.printing as printing
 
+# A token value Python's tokenizer probably will never use.
+_BACKTICK_QUOTED_STRING = 100
+
+Tok = Tuple[int, str]
+
+
+def _create_valid_python_identifier(name: str) -> str:
+    """Create valid Python identifiers from any string.
+
+    Check if name contains any special characters. If it contains any
+    special characters, the special characters will be replaced by a special
+    string and a prefix is added.
+
+    Raises
+    ------
+    If the returned name is not a Python valid identifier, raise an exception.
 
-def tokenize_string(source):
     """
-    Tokenize a Python source code string.
+    if name.isidentifier() and not iskeyword(name):
+        return name
+
+    # Create a dict with the special characters and their replacement.
+    # To create the dict, we loop over all EXACT_TOKEN_TYPES with a single character
+    # (Multiple character operators will be replaced as well because they are
+    # composed of single character operators.)
+    # and take the character alongside with its human readable description from
+    # token.tok_name.
+    # This description is surrounded with underscores for better
+    # readability in the final name.
+    # Also added some more characters, that are not operators,
+    # that people might use in column names.
+    special_characters_replacements = {
+        " ": "_",
+        "?": "_QUESTIONMARK_",
+        "!": "_EXCLAMATIONMARK_",
+        "$": "_DOLLARSIGN_",
+        "€": "_EUROSIGN_",
+        # Including quotes works, but they have to come in pairs.
+        "'": "_SINGLEQUOTE_",
+        '"': "_DOUBLEQUOTE_",
+        # Currently not possible. Terminates parser and won't find backtick.
+        # "#": "_HASH_",
+        **{
```

Comment:
Can you create this dict separately and then just call `special_characters_replacements.update(...` on it? I think easier to read than the argument unpacking here